compiler/rustc_mir_transform/src/coroutine.rs

```diff
@@ -1221,6 +1298,21 @@ fn elaborate_coroutine_drops<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
         );
     }
     elaborator.patch.apply(body);
+    for block_data in body.basic_blocks_mut() {
```

Comment:
Is there some code up above this (or called by one of the methods above) that was in principle responsible for dropping the upvars, that is now made obsolete under the new regime you have introduced here?